src/python/WMCore/MicroService/Unified/MSOutputTemplate.py

```diff
@@ -0,0 +1,228 @@
+import time
+
+from copy import deepcopy
+from bson import Timestamp, ObjectId
+
+
+class MSOutputTemplate(dict):
+    """
+    A simple template to represent the objects stored in Mongodb for MSOutput
+    Its purpose should be to serve as a bidirectional representation of the
+    objects stored in Mongodb. The instances of this class should be used as a
+    buffer for both inbound (reading from the database) and outbound (writing to
+    database) documents. And it must have all the methods needed to set or update
+    various fields in the object. So that we assure the uniformity of all documents.
+
+    Template format:
+    {
+    "RequestName": "ReqName",
+    "RequestStatus": "(clompletd|cloed-out|announced),
+    "creationTime": integer timestamp,
+    "lastUpdate": integer timestamp,
+    "isRelVal": (True|False),
+    "isTaken": (True|False),
+    "destination": ["list of locations"],
+    "OutputDatasets": ["list of datasets"],
+    "destinationOutputMap": [{"destination": ["list of locations"],
+                              "datasets": ["list of datasets"]},
+                             {"destination": ["list of locations"],
+                              "datasets": ["list of datasets"]}],
+    "campaignOutputMap": [{"campaignName": "blah",
+                           "datasets": ["list of datasets"]},
+                          {"campaignName": "blah",
+                           "datasets": ["list of datasets"]}],
+    "transferStatus": "blah", # either "pending" or "done",
+    "transferIDs": ["list of transfer IDs"],
+    "numberOfCopies": integer
+    }
+    """
+    # TODO:
```

Comment:
I think we can default the id to request['RequestName'], fail if such key does not exist.

Reply:
It happens automatically because of the piece of code I just pasted in the previous comment